src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3572,6 +3572,11 @@ private IOperation VisitNoneOperationExpression(IOperation operation)
             return Operation.CreateOperationNone(semanticModel: null, operation.Syntax, operation.ConstantValue, childrenBuilder.ToImmutableAndFree(), operation.IsImplicit);
         }
 
+        public override IOperation VisitNameOf(INameOfOperation operation, int? captureIdForResult)
+        {
+            return new NameOfExpression(Visit(operation.Argument), semanticModel: null, operation.Syntax, operation.Type, operation.ConstantValue, operation.IsImplicit);
```

Comment:
>Visit(operation.Argument) [](start = 40, length = 25)

I think we should simply replace this node with a string literal that it evaluates to. Visiting the argument is not the right thing to do, no matter what is there, it is not going to be executed and should any control flow be inside this expression, this should not have any effect on the flow graph.